### PR TITLE
Java and Ruby bug fixes, favicon and connection status support

### PR DESCRIPTION
**Pull request by ldilley**
## Proposed Changes
- Add favicon support (related to #120).
- Expose connection status (related to #119).
- Increment Java version from `3.0.1` to `3.0.2`.
- Increment Ruby version from `2.2.1` to `2.2.2`.
- Fix bug in Java version during MotD strip where JSON query `description` field was a string instead of a JSON object.
- Fix bug where connection details were overwritten if a Bedrock request was attempted after a JSON request was successful.

### CODE REVIEW

File: Ruby/lib/minestat.rb [hunk 1]
```diff
@@ -79,65 +80,77 @@
~
~  ##
~  # Instantiate an instance of MineStat and poll the specified server for information
~  def initialize(address, port = DEFAULT_TCP_PORT, timeout = DEFAULT_TIMEOUT, request_type = Request::NONE)
```
[the remark this answers is not in the document]
Comment (ldilley) [reply Ruby/lib/minestat.rb]: @sonatype-lift ignore

**Comment by ldilley** [comment]
No worries at all, @mindsolve -- I've also been delayed due to a battle with the flu. Thank you for catching the connection status being overwritten and for providing a 1.3 server to test against. There were actually 2 other issues pertaining to data being overwritten (request type and online status if Bedrock was being checked). These problems have all been corrected in the Ruby implementation. I still need to review the Java code. I've incremented the Ruby version again since I recently published `2.2.2` to RubyGems.

For the connection status data type, this was an attempt at making the information more user friendly similar to what `perror()` or `strerror()` do. I can still store the enum value which is essentially an integer I believe. However, the developer would need to refer to the source code to see what the return codes mean.

**Comment by ldilley** [comment]
Not sure if you saw my note in Discord, @mindsolve -- This PR is ready for re-review. Let me know if you're strongly opposed to using strings for the connection status rather than enum values. I do not have any qualms about changing the return type if so.

**Review by mindsolve — APPROVED** [review]
PR looks great! Very nice :)
Thanks a lot for the great work!

**Pull request merged**